NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -187,6 +207,105 @@ public GroupPermission getPermissionforGroup(Group group){
 		return permhandle.getGroupPermission(group);
 	}
 	
+	// == GROUP INTERACTIONS ============================================================== //
+	
+	public List<UUID> getMembers(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) { 
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers();
+	}
+	
+	public List<UUID> getMembers(String groupname, PlayerType rank) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return Lists.newArrayList();
+		}
+		return group.getAllMembers(rank);
+	}
+	
+	public PlayerType getMemberRank(String groupname, UUID player) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return null;
+		}
+		return group.getMember(player);
+	}
+	
+	public boolean isMember(String groupname, UUID player) {		
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		return group.isMember(player);
+	}
+	
+	public boolean hasAccess(String groupname, UUID player, PermissionType perm) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return false;
+		}
+		
+		GroupPermission perms = getPermissionforGroup(group);
+		PlayerType rank = group.getMember(player);	
+		
+		return perms.isAccessible(rank, perm);
+	}
+	
+	/**
+	 * @param groupname - String name of the group
+	 * @return false if the group can be operated on, true otherwise
+	 */
+	public boolean isDisciplined(String groupname) {
+		Group group = getGroup(groupname);
+		if (group == null) {
+			return true;
+		}
+		
+		return group.isDisciplined();
+	}
+	
+	public boolean areLinked(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		return Group.areLinked(supergroup, subgroup);
+	}
+	
+	public boolean link(String supername, String subname) {
+		Group supergroup = getGroup(supername);
+		Group subgroup = getGroup(subname);
+		if (supergroup == null || subgroup == null) {
+			return false;
+		}
+		
+		boolean success = Group.link(supergroup, subgroup);
+		if (success) {
+			groupManagerDao.addSubGroup(supername, subname);
```

Comment:
That will completly recache the group, technically it should work. Other stuff only has it's own message so other shards can notify their plugins about what happened, for example when a groups get merged the other shards get a special message for that, because JukeAlert needs to update snitch groups.


If all of the permission stuff is working right you shouldnt have to notify other plugins of linking/unlinking ever and everything would just get handled over NameLayers API, as long as thats the case you should be good.